pifpaf/drivers/__init__.py

```diff
@@ -91,18 +91,23 @@ def _ensure_xattr_support(self):
                                self.__class__.__name__)
 
     def _kill(self, process):
-        if not process.is_alive():
-            return
-        pgrp = os.getpgid(process.pid)
-        process.terminate()
-        try:
-
-            self._wait(process)
-        except tenacity.RetryError:
-            LOG.warning("PID %d didn't terminate cleanly after 10 seconds, "
-                        "sending SIGKILL to its process group", process.pid)
+        ret = process.poll()
+        if ret is None:
+            process.terminate()
```

Comment:
This can lead to a race condition if the process dies between `process.poll()` and `process.terminate()`.